ocean_visual.cc

```diff
@@ -107,65 +105,121 @@ namespace asv
     // OGRE setup
     this->data->scene = _visual->GetScene();
 
+    // Only load plugin once
+    // load the visual plugin in gzclient for now.
+    if (!this->data->scene->EnableVisualizations())
+      return;
+
+    // Setup camera
+    Ogre::Camera *userCamera = this->data->scene->GetUserCamera(0)->OgreCamera();
+    if (userCamera)
+    {
+      this->data->trackedCamera = userCamera;
+      this->data->camera =
```

Comment:
I tried this a few days ago, but had issues. For some reason, when I set `camera->setPosition(user_camera->getPosition())` and `camera->setOrientation(user_camera->getOrientation())`, it was completely off. I tested this by having a miniscreen (Ogre::Rectangle2d) that displayed the texture created from the new camera, but it was looking completely off. I am not sure if that issue is relevant here.

Also, I got confused because this->data->scene->GetUserCamera(0)->WorldRotation() != this->data->scene->GetUserCamera(0)->OgreCamera()->getDerivedOrientation(). Is this expected?

Reply:
Is there a reason why you need to make the new camera? I tried it out, but it lead to more issues than I saw benefits.

Reply:
`user_camera->getPosition()` and `user_camera->getOrientation()` gets the local position and orientation relative to the parent but we need to the derived position and orientation instead

It is expected that `WorldRotation != OgreCamera()->getDerivedOrientation`. It's a little confusing but `WorldRotation` returns the camera's parent node's world rotation, not the actual camera itself, i.e. `WorldRotation = OgreCamera()->getParentSceneNode()->getDerivedOrientation()`

 The camera object is rotated to be z-up when it's created: https://bitbucket.org/osrf/gazebo/src/2b59492f96fcf121b33f290fd5cf0b70b42f4e22/gazebo/rendering/Camera.cc#lines-1501

Having said all this, I think the new camera is probably not needed. We can remove it to simplify things a bit. I added it because I thought we needed to modify some properties and but looks like it's not needed.

Reply:
Regarding the dark plane issue. I'm guessing it's because I forgot to set the viewport's background color. It defaults to black so it's probably reflecting the black background color. I need to verify this.

Reply:
I tried adding `vp->setBackgroundColour(Ogre::ColourValue::White);` but it is still Black. This is very strange.